tests/s3/copy_object/test_parallel_copy_object.py

```diff
@@ -365,3 +381,140 @@ def test_44793(self):
+                                                f"{obj_type}-obj1"),
+                                                numproc=2,
+                                                assert_flag=False)
+            except CTException as error:
+                LOGGER.info(error.message)
+                assert_utils.assert_in(errmsg.NO_SUCH_KEY_ERR,error.message, error)
+            LOGGER.info("List %s-obj2 to dest-bucket1", obj_type)
+            self.objects_in_buckets(bucket="src-bucket1", objects=[f"{obj_type}-obj1"])
+            self.objects_in_buckets(bucket="dest-bucket1", objects=[f"{obj_type}-obj2"])
+            self.parallel_copy_object(("src-bucket1", f"{obj_type}-obj1", "dest-bucket1",
+                                      f"{obj_type}-obj2"),
+                                      ("dest-bucket1", f"{obj_type}-obj2", "src-bucket1",
+                                      f"{obj_type}-obj1"))
+            resp = self.s3_obj.object_info("dest-bucket1", f"{obj_type}-obj2")
+            assert_utils.assert_equal(resp1[1]["ETag"], resp[1]["ETag"])
+            self.parallel_copy_object(("src-bucket1", f"{obj_type}-obj1", "dest-bucket1",
+                                      f"{obj_type}-obj2"),
+                                      ("dest-bucket1", f"{obj_type}-obj3", "src-bucket1",
+                                      f"{obj_type}-obj4"))
+            resp = self.s3_obj.object_info("dest-bucket1", f"{obj_type}-obj2")
+            assert_utils.assert_equal(resp1[1]["ETag"], resp[1]["ETag"])
+            src_resp = self.s3_obj.object_info("src-bucket1", f"{obj_type}-obj4")
+            assert_utils.assert_equal(resp2[1]["ETag"], src_resp[1]["ETag"])
+
+            self.multi_parallel_copy_object(("dest-bucket1", f"{obj_type}-obj2", "src-bucket1",
+                                            f"{obj_type}-obj3"),
+                                            ("dest-bucket1", f"{obj_type}-obj2", "src-bucket1",
+                                            f"{obj_type}-obj4"),
+                                            ("dest-bucket1", f"{obj_type}-obj2", "dest-bucket2",
+                                            f"{obj_type}-obj4"),
+                                            numproc=3)
+            dest_resp = self.s3_obj.object_info("dest-bucket1", f"{obj_type}-obj2")
+            src_resp = self.s3_obj.object_info("src-bucket1", f"{obj_type}-obj3")
+            assert_utils.assert_equal(dest_resp[1]["ETag"], src_resp[1]["ETag"])
+            src_resp = self.s3_obj.object_info("src-bucket1", f"{obj_type}-obj4")
+            assert_utils.assert_equal(dest_resp[1]["ETag"], src_resp[1]["ETag"])
+            dest_resp1 = self.s3_obj.object_info("dest-bucket2", f"{obj_type}-obj4")
+            assert_utils.assert_equal(dest_resp[1]["ETag"], src_resp[1]["ETag"])
+        LOGGER.info("ENDED: Test Reverse copy operations (simple and multipart source objects)")
```

Comment:
last step 15, listing objets in buckets is not implemented here

Reply:
in a source and destination bucket , we need to check which object are present and are they valid as we created/copied to specific bucket . I think Ketan already implemented it in last sprint, you can reuse it

Reply:
updated